main/src/ca/uwaterloo/flix/language/ast/ErasedAst.scala

```diff
@@ -216,6 +206,24 @@ object ErasedAst {
 
   }
 
+  sealed trait IntrinsicOperatorN
+
+  object IntrinsicOperatorN {
+
+    case class Closure(sym: Symbol.DefnSym) extends IntrinsicOperatorN
+
+    case class ApplyClo(exp: ErasedAst.Expression) extends IntrinsicOperatorN
```

Comment:
One thing that is important is that Intrinsics should not contain expression themselves. The point is to keep expression-related stuff in expression and have Intrinsics only contain some meta data.

Reply:
If you look at ApplyClo I think it should really be represented as new intrinsic: `IntrinsicOperator1N` which has the expression:

```
    case class IntrinsicN(op: ErasedAst.IntrinsicOperatorN, exp: Expression, exps: List[ErasedAst.Expression], tpe: MonoType, loc: SourceLocation) extends Expression
```